src/coreclr/nativeaot/Runtime/gcrhenv.cpp

```diff
@@ -1178,7 +1178,7 @@ void GCToEEInterface::StompWriteBarrier(WriteBarrierParameters* args)
 
 void GCToEEInterface::EnableFinalization(bool foundFinalizers)
 {
-    if (foundFinalizers)
+    if (foundFinalizers || (GCHeapUtilities::GetGCHeap()->GetNoGCFinalizerAction(false) != finalizer_action_kind::none))
```

Comment:
Instead of calling back to GC to check whether there is work to do, we can rename `foundFinalizers` to `gcHasWorkForFinalizerThread`.